kasa/device_factory.py

```diff
@@ -167,7 +167,7 @@ def get_device_class_from_sys_info(sysinfo: dict[str, Any]) -> type[IotDevice]:
 
 
 def get_device_class_from_family(
-    device_type: str, *, https: bool
+    device_type: str, *, https: bool, no_warn: bool = False
```

Comment:
`silent` or `quiet`?

Reply:
Renamed to `require_exact` as it doesn't actually return the default in this instance either.